Bottom: 682fd858497b379d36f503b41d6d16454d86b9f9
Top:    66307aef81cd8bb1ffe59b40664a94e05b802129
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-29 16:57:43 +0100

Refresh of hook-up-debug-drawing-apis-for

---

diff --git a/src/hid.h b/src/hid.h
index 2b9ca40..08ec71f 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -230,9 +230,10 @@ typedef enum
     int (*throw_drc_dialog) (void);
   } HID_DRC_GUI;
 
+  typedef struct hid_st HID;
 
 /* This is the main HID structure.  */
-  typedef struct
+  struct hid_st
   {
     /* The size of this structure.  We use this as a compatibility
        check; a HID built with a different hid.h than we're expecting
@@ -543,7 +544,11 @@ typedef enum
 
     void (*edit_attributes) (char *owner, AttributeListType *attrlist_);
 
-  } HID;
+    HID *(*request_debug_draw) (void);
+    void (*flush_debug_draw)   (void);
+    void (*finish_debug_draw)  (void);
+
+  };
 
 /* Call this as soon as possible from main().  No other HID calls are
    valid until this is called.  */
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index c146fd2..f9d5949 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -427,6 +427,22 @@ nogui_progress (int so_far, int total, const char *message)
   return 0;
 }
 
+static HID *
+nogui_request_debug_draw (void)
+{
+  return NULL;
+}
+
+static void
+nogui_flush_debug_draw (void)
+{
+}
+
+static void
+nogui_finish_debug_draw (void)
+{
+}
+
 HID hid_nogui = {
   sizeof (HID),
   "nogui",
@@ -484,7 +500,10 @@ HID hid_nogui = {
   nogui_beep,
   nogui_progress,
   0 /* nogui_drc_gui */ ,
-  0 /* edit_attributes */
+  0 /* edit_attributes */,
+  nogui_request_debug_draw,
+  nogui_flush_debug_draw,
+  nogui_finish_debug_draw,
 };
 
 #define AD(x) if (!d->x) d->x = s->x
@@ -545,4 +564,7 @@ apply_default_hid (HID * d, HID * s)
   AD (progress);
   AD (drc_gui);
   AD (edit_attributes);
+  AD (request_debug_draw);
+  AD (flush_debug_draw);
+  AD (finish_debug_draw);
 }
diff --git a/src/hid/gtk/gtkhid-gdk.c b/src/hid/gtk/gtkhid-gdk.c
index 27d62c6..b5e28b6 100644
--- a/src/hid/gtk/gtkhid-gdk.c
+++ b/src/hid/gtk/gtkhid-gdk.c
@@ -1226,3 +1226,17 @@ ghid_render_pixmap (int cx, int cy, double zoom, int width, int height, int dept
 
   return pixmap;
 }
+
+HID *
+ghid_request_debug_draw (void)
+{
+}
+
+void
+ghid_flush_debug_draw (void)
+{
+}
+
+ghid_finish_debug_draw (void)
+{
+}
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index f5e2717..ac961db 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -2157,6 +2157,10 @@ hid_gtk_init ()
   ghid_hid.drc_gui                  = &ghid_drc_gui,
   ghid_hid.edit_attributes          = ghid_attributes;
 
+  ghid_hid.request_debug_draw       = ghid_request_debug_draw;
+  ghid_hid.flush_debug_draw         = ghid_flush_debug_draw;
+  ghid_hid.finish_debug_draw        = ghid_finish_debug_draw;
+
   hid_register_hid (&ghid_hid);
 #include "gtk_lists.h"
 }
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 6cdeff2..d52d088 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -499,6 +499,9 @@ gboolean ghid_drawing_area_expose_cb (GtkWidget *, GdkEventExpose *,
 gboolean ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * ev);
 GdkPixmap *ghid_render_pixmap (int cx, int cy, double zoom,
                                int width, int height, int depth);
+HID *ghid_request_debug_draw (void);
+void ghid_flush_debug_draw (void);
+void ghid_finish_debug_draw (void);
 
 /* gtkhid-main.c */
 void ghid_pan_fixup (void);
